core/src/main/java/software/amazon/awssdk/retry/conditions/MaxNumberOfRetriesCondition.java

```diff
@@ -25,7 +25,7 @@
     private final int maxNumberOfRetries;
 
     public MaxNumberOfRetriesCondition(int maxNumberOfRetries) {
-        this.maxNumberOfRetries = assertIsPositive(maxNumberOfRetries, "maxNumberOfRetries");
+        this.maxNumberOfRetries = maxNumberOfRetries;
```

Comment:
Are we okay with negatives now?

Reply:
Yes. We can add this back but not sure we need to. Anything  <= 0 will just be 0 retries.

Reply:
meh I'd rather not do that. Lots of contracts use -1 or something to mean infinite and I'd rather not open ourselves up to that confusion. We can always relax constraints in the future if there is a need.